src/services/textChanges.ts

```diff
@@ -727,49 +719,24 @@ namespace ts.textChanges {
                     // a, b, c
                     // create change for adding 'e' after 'a' as
                     // - find start of next element after a (it is b)
-                    // - use this start as start and end position in final change
-                    // - build text of change by formatting the text of node + separator + whitespace trivia of b
+                    // - use next element start as start and end position in final change
+                    // - build text of change by formatting the text of node + whitespace trivia of b
 
                     // in multiline case it will work as
                     //   a,
                     //   b,
                     //   c,
                     // result - '*' denotes leading trivia that will be inserted after new text (displayed as '#')
-                    //   a,*
-                    // ***insertedtext<separator>#
+                    //   a,
+                    //   insertedtext<separator>#
                     // ###b,
                     //   c,
-                    // find line and character of the next element
-                    const lineAndCharOfNextElement = getLineAndCharacterOfPosition(sourceFile, skipWhitespacesAndLineBreaks(sourceFile.text, containingList[index + 1].getFullStart()));
-                    // find line and character of the token that precedes next element (usually it is separator)
-                    const lineAndCharOfNextToken = getLineAndCharacterOfPosition(sourceFile, nextToken.end);
-                    let prefix: string | undefined;
-                    let startPos: number;
-                    if (lineAndCharOfNextToken.line === lineAndCharOfNextElement.line) {
-                        // next element is located on the same line with separator:
-                        // a,$$$$b
-                        //  ^    ^
-                        //  |    |-next element
-                        //  |-separator
-                        // where $$$ is some leading trivia
-                        // for a newly inserted node we'll maintain the same relative position comparing to separator and replace leading trivia with spaces
-                        // a,    x,$$$$b
-                        //  ^    ^     ^
-                        //  |    |     |-next element
-                        //  |    |-new inserted node padded with spaces
-                        //  |-separator
-                        startPos = nextToken.end;
-                        prefix = spaces(lineAndCharOfNextElement.character - lineAndCharOfNextToken.character);
-                    }
-                    else {
-                        // next element is located on different line that separator
-                        // let insert position be the beginning of the line that contains next element
-                        startPos = getStartPositionOfLine(lineAndCharOfNextElement.line, sourceFile);

```

Comment:
There are a couple failing tests now; I’m guessing some of this was still important. If I had to guess, I would say that this `else` block needs to be preserved for insertions where the next element is on a different line.

Reply:
I really thought I have run the tests. Anyway is fixed now. We still don't need all of that we just need to consider the trivia.